Sources/Subs-Calendar.php

```diff
@@ -1863,4 +1863,40 @@ function removeHolidays($holiday_ids)
 	));
 }
 
+/**
+ * Helper function to convert date string to english
+ * so that date_parse can parse the date
+ *
+ * @param string $date A localized date string
+ * @return string English date string
+ */
+function convertDateToEnglish($date)
+{
+	global $txt, $context;
+
+	if ($context['user']['language'] == 'english')
+		return $date;
+
+	$replacements = array_combine(array_map('strtolower', $txt['months_titles']), array(
+		'January', 'February', 'March', 'April', 'May', 'June',
+		'July', 'August', 'September', 'October', 'November', 'December'
+	));
+	$replacements += array_combine(array_map('strtolower', $txt['months_short']), array(
+		'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
+		'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'
+	));
+	$replacements += array_combine(array_map('strtolower', $txt['days']), array(
+		'Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'
+	));
+	$replacements += array_combine(array_map('strtolower', $txt['days_short']), array(
+		'Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'
+	));
+	$replacements[strtolower($txt['time_am'])] = 'AM';
+	$replacements[strtolower($txt['time_pm'])] = 'PM';
+
+	$replacements = array_map('strtolower', $replacements);
+
+	return strtr(strtolower($date), $replacements);
```

Comment:
1. Why `strtolower()`?
2. In order to catch all possible variants of `AM` and `PM` for the current language, I suggest the following. It might seem redundant, but it is necessary in order to catch all possible outcomes of the `timeformat()` function.
```suggestion
	// Find all possible variants of AM and PM for this language.
	$replacements[$txt['time_am']] = 'AM';
	$replacements[$txt['time_pm']] = 'PM';
	if (($am = strftime('%p', strtotime('01:00:00'))) !== 'p' && $am !== false)
	{
		$replacements[$am] = 'AM';
		$replacements[strftime('%p', strtotime('23:00:00'))] = 'PM';
	}
	if (($am = strftime('%P', strtotime('01:00:00'))) !== 'P' && $am !== false)
	{
		$replacements[$am] = 'AM';
		$replacements[strftime('%P', strtotime('23:00:00'))] = 'PM';
	}

	return strtr($date, $replacements);
```

Reply:
1. Yes, you are right. No point in converting the values.
2. Since we convert it lower case. Do we need to check both %P and %p? Aren't those just different cases?